contracts/validator-manager/SlotAuctionManager.sol

```diff
@@ -5,242 +5,440 @@
             revert AuctionInProgress();
         }
-        if (auctionState == AuctionState.AuctionFinalizing) {
+        if ($._auctionState == AuctionState.AuctionFinalizing) {
             revert AuctionFinalizing();
         }
         _;
     }
 
-    // TODO replace this ^ with this v once I know it wont cause issues for me
-
-    // function initialize(address tokenAddress, address vmAddress)public initializer {
-    //     TOKEN_CONTRACT = IERC20(tokenAddress);
-    //     VALIDATOR_MANAGER = IValidatorManager(vmAddress);
-    // }
-
-    
-    function initiateAuction(
-        uint16 validatorslots,
-        uint64 weight,
-        uint256 minAuctionDuration,
-        uint256 minValidatorDuration,
-        uint256 minimumBid
-    ) AuctionOff external {
-        auctionState = AuctionState.AuctionInProgress;
-        validatorWeight = weight;
-        // require(churn percentage will not be overlooked)
-        
-        auctionEndTime = block.timestamp + minAuctionDuration; 
-        validatorSlots = validatorslots;
-        MinValidatorDuration = minValidatorDuration;
-        MinimumBid = minimumBid;
-        _secondPrice = 0;
-        // Initiates an empty heap array, id rather just create a new one but solidity memory syntax is still confusing to me
-        delete _bids.tree; 
-        emit NewValidatorAuction(validatorSlots, weight, minValidatorDuration, minAuctionDuration, MinimumBid);
+    modifier AuctionCooldown() {
+        SlotAuctionManagerStorage storage $ = _getSlotAuctionManagerStorage();
+
+        if (block.timestamp < $._auctionCooldownEndtime) {
+            revert AuctionInCooldown($._auctionCooldownEndtime);
+        }
+        _;
     }
 
+    // solhint-disable-next-line func-name-mixedcase
+    function __SlotAuctionManager_init(
+        SlotAuctionManagerSettings calldata settings
+    ) internal onlyInitializing {
+        __ReentrancyGuard_init();
+        __Ownable_init(settings.admin);
+        __SlotAuctionManager_init_unchained(settings.manager, settings.auctionSettings);
+    }
 
-    function endAuction() nonReentrant AuctionOn external {
+    // solhint-disable-next-line func-name-mixedcase
+    function __SlotAuctionManager_init_unchained(
+        IValidatorManager manager,
+        AuctionSettings calldata auctionSettings
+    ) internal onlyInitializing {
+        SlotAuctionManagerStorage storage $ = _getSlotAuctionManagerStorage();
 
-        auctionState = AuctionState.AuctionFinalizing;
-        if (block.timestamp <= auctionEndTime) {
-            revert AuctionEndTimeNotReached(auctionEndTime);
+        if (address(manager) == address(0)) {
+            revert ZeroAddress();
+        }
+        // Minimum stake duration should be at least one churn period in order to prevent churn tracker abuse.
+        if (auctionSettings.minValidatorDuration < manager.getChurnPeriodSeconds()) {
+            revert InvalidMinValidatorDuration(auctionSettings.minValidatorDuration);
+        }
```

Comment:
I don't think we need this check with the slot auction manager... This constant was introduced into the staking managers so that someone could not stake and unstake rapidly to fill up the churn tracker space. The minimum duration is just how long they have to wait to get their stake back after initially registering, basically